source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -1956,38 +1956,19 @@ bool SwiftLanguageRuntime::GetDynamicTypeAndAddress_Promise(
     if (swift_ast_ctx->IsErrorType(protocol_type)) {
```

Comment:
Out of curiosity: Do you understand why error is handled specially here? I'd be curious to learn why we can't handle all existential types the same way.

Reply:
Error has a special, NSError-compatible layout.

Reply:
Thanks, that makes sense!

Reply:
It might be worth looking at the LLVM IR generated for the testcase on various platforms then, to check whether any of the contortions performed by the old code perhaps make sense on any other targets (like i386 for example).

Reply:
Checked, it looks fine.